### ISSUE TITLE
omd 2.0.0: regression with URLs containing spaces

### ISSUE THREAD

**Issue by Armael** [ocaml/omd#237]
I'm observing a regression with omd 2.0.0 compared to omd 1.3.1 (it broke the rendering of some webpage I have). When parsing a URL with a file name containing a space:

```
[foo](http://site.com/a b.pdf)
```

omd 3.1 will parse it as a URL: 
```[Omd.Paragraph [Omd.Url ("http://site.com/a b.pdf", [Omd.Text "foo"], "")]]```

omd 2.0.0 will parse it as text:
```
[{Omd.bl_desc =
   Omd.Paragraph
    {Omd.il_desc =
      Omd.Concat
       [{Omd.il_desc = Omd.Text "["; il_attributes = []};
        {Omd.il_desc = Omd.Text "foo"; il_attributes = []};
        {Omd.il_desc = Omd.Text "](http://site.com/a b.pdf)";
         il_attributes = []}];
     il_attributes = []};
  bl_attributes = []}]
```

**Comment by SquidDev** [comment]
[Trying against the cmark dingus](https://spec.commonmark.org/dingus/?text=%5Bfoo%5D(http%3A%2F%2Fsite.com%2Fa%20b.pdf)) seems to suggest that 2.0's behaviour is correct. [The spec][spec] does say the following (emphasis mine):

> A link destination consists of either [..] a nonempty sequence of characters that does not start with <, _does not include ASCII space_ or control characters.

[spec]: https://spec.commonmark.org/0.29/#link-destination

**Comment by Armael** [comment]
ah, thanks, that's good to know! Should the fix then to replace spaces with `%20` in the filename?

**Comment by SquidDev** [comment]
Yes, either of these should work:

```
[foo](http://site.com/a%20b.pdf)
[foo](<http://site.com/a b.pdf>)
```

**Comment by Armael** [comment]
Oh, nice. Thanks again for the help, and happy to know that the new behavior is the correct one :).

**Comment by Armael** [comment]
Well, it looks like omd does not handle the nicer syntax (the one with the `<>`). :-(

**Comment by SquidDev** [comment]
I _think_ that's fixed in #233, but currently waiting on another PR to be merged before that one is.

**Comment by shonfeder** [comment]
Thanks for the report, @Armael and for fielding the response, @SquidDev! 

Let's keep this open for now, since others might hit his problem while the alpha is up, and we should highlight this breaking change when we do do the 2.0 release.

**Comment by shonfeder** [comment]
Confirmed this was fixed in #233. Thanks!
